models/Transaction.go

```diff
@@ -142,3 +148,16 @@ func GetAskTransactionsForStock(stockID, count uint32) ([]*Transaction, error) {
 	l.Debugf("Done")
 	return transactions, nil
 }
+
+// GetTransactionRef creates and returns a reference of a Transaction
+func GetTransactionRef(userID, stockID uint32, ttype TransactionType, qty int64, price uint64, total int64, createdAt string) *Transaction {
```

Comment:
If all the function does is return a struct, why have a function? [If you're trying to avoid writing struct field names, that's already allowed, although not recommended].

You can have a function if it fills defaults sensibly, like `CreatedAt` etc.

Reply:
It sort of looked cleaner to just create a function so that the main function becomes a bit more english-like. But yeah, this function doesn't really add any additional value apart from that.